### DISCUSSION TITLE
lutris: gamemode does not work if using startx instead of login manager

### DISCUSSION THREAD

**Discussion by pinbuck** [netblue30/firejail#5351] [Q&A]
I put this in discussions because I don't believe its a bug with firejail, but a problem I'm trying to solve.
I assume startx is not the problem here because gamemode under lutris is the only problem I have. would starting gamemoded

This is *only* when I don't have a login manager, and am using startx. --net=none does not work like it usually does.

For context, I am simply launching firejail with no default profile, and just to block internet to an application and assumedly all of their spawned child processes, such as gamemode, mangohud, the wine prefix itself, and whatever is running in that prefix. I can switch to SDDM display/login manager anytime to make this issue not apparent, but that isnt desired.

I posted a comment asking about deleting gamemode's group itself and then reinstalling it and having the script do it instead. I believe I messed up gamemode's group when I did those commands, and it's likely the culprit of this entire problem:
https://github.com/FeralInteractive/gamemode/issues/351

Everything is normal when running lutris alone instead of adding anything before it:
$ lutris
[log lutris normal.txt](https://github.com/netblue30/firejail/files/9449978/log.lutris.normal.txt)

works and creates about the same log lines, but then there is no real benefit of firejail:
$ firejail --noprofile lutris
[log lutris firejail no profile.txt](https://github.com/netblue30/firejail/files/9450096/log.lutris.firejail.no.profile.txt)

But when I try to run games under firejail with gamemode enabled:
$ firejail --net=none --noprofile lutris
[log lutris firejail.txt](https://github.com/netblue30/firejail/files/9449974/log.lutris.firejail.txt)

Running without gamemode enabled:
$ firejail --net=none --noprofile lutris
[firejail lutris no gamemode.txt](https://github.com/netblue30/firejail/files/9450118/firejail.lutris.no.gamemode.txt)

**Comment by ghost** [comment]
Going over the discussion items I noticed this hasn't received a response (yet). In that vain, please note this isn't an answer in the sense that it would contain a fix or workaround for your stated problem. I just wanted to point out that using `--noprofile` is a Firejail option that needs to be handled with care. It's a debug thing more than anything else really, and it can provide that false sense of security if a user assumes things get properly sandboxed in some way just by using the option. Hope you can get eyes on what you're trying to achieve, best of luck.

**Reply by pinbuck** [reply]
> Going over the discussion items I noticed this hasn't received a response (yet). In that vain, please note this isn't an answer in the sense that it would contain a fix or workaround for your stated problem. I just wanted to point out that using `--noprofile` is a Firejail option that needs to be handled with care. It's a debug thing more than anything else really, and it can provide that false sense of security if a user assumes things get properly sandboxed in some way just by using the option. Hope you can get eyes on what you're trying to achieve, best of luck.

I use
"firejail --net=none --noprofile"
just becase I have yet to find other easy and/or reliable solutions. as soon as I'm able to reliably block internet for a process and its child processes. I was simply looking for a lighter option than opensnitch. I've considered using opensnitch again but I dont really want to have gui tools running all the time and/or at startup.
I fully understand that --noprofile makes firejail not use an application's profile, rendering firejail to do nothing.
https://askubuntu.com/questions/19346/how-to-block-internet-access-for-wine-applications
I did the no-net group thing above, but I think it had issues with certain applications. I'm not sure I remember.

I solved my issue by moving the command to the command prefix of the game, everything works as it should. Only disapointing thing is "gamemode on" for mangohud's in-game hud won't display. I've encountered this issue before, but it's not too concerning.
![image](https://user-images.githubusercontent.com/96911866/194458496-51b1126b-fa10-45ba-80a5-e29db2f5e59e.png)

before I close this, do you know how to simply link just one singular application alone to its firejail profile?
I want to simply put "firejail application" instead of the longer above command, but lutris doesn't seem to link a profile for itself at all.
![image](https://user-images.githubusercontent.com/96911866/194459432-6fb246c6-9ebb-4204-b103-255fa7d1a364.png)
